drivers/vulkan/vulkan_context.cpp

```diff
@@ -400,16 +400,26 @@ Error VulkanContext::_obtain_vulkan_version() {
 	return OK;
 }
 
-Error VulkanContext::_initialize_extensions() {
-	uint32_t instance_extension_count = 0;
+HashMap<CharString, bool> VulkanContext::requested_instance_extensions;
+
+void VulkanContext::register_requested_instance_extension(const CharString &extension_name, bool p_required) {
```

Comment:
Right now this has to be called before Vulkan is initialized. Do we want this to set a dirty flag that would require Vulkan to re-initialize? In other words, is there ever a situation where we might enable an extension after drawing has commenced?

Reply:
My opinion is that this should not be allowed, recreating the Vulkan instance would basically be a restart of the engine. Maybe we should add a check to register_requested_instance_extension so it errors if the instance has already been created?

Reply:
Hmm, we don't have singleton access or anything and it seems overkill to add it just for this.

Reply:
After talking with Clay just add some boolean states for this.